quark/core/shurikenapkinfo.py

```diff
@@ -0,0 +1,403 @@
+
+        :param instruction: instruction instance from androguard
+        :return: a list with bytecode instructions strings
+        """
+        pass
+
+    def _find_first_bytecode_by_calling_method(
+        self, bytecodes: Iterator[BytecodeObject], target_method: MethodObject
+    ) -> Optional[BytecodeObject]:
+        targetMethodCall = f"{target_method.class_name}->{target_method.name}{target_method.descriptor}"
+
+        for bytecode in bytecodes:
+            if (
+                bytecode.mnemonic.startswith("invoke")
+                and targetMethodCall in bytecode.parameter
+            ):
+                return bytecode
+
+    @functools.lru_cache()
+    def get_wrapper_smali(
+        self,
+        parent_method: MethodObject,
+        first_method: MethodObject,
+        second_method: MethodObject,
+    ) -> Dict[str, Union[BytecodeObject, str]]:
+        bytecodes = self.get_method_bytecode(parent_method)
+
+        first = self._find_first_bytecode_by_calling_method(
+            bytecodes, first_method
+        )
+        second = self._find_first_bytecode_by_calling_method(
+            bytecodes, second_method
+        )
+
+        return {
+            "first": [
+                first.mnemonic,
+                " ".join(first.registers),
+                first.parameter,
+            ],
+            "first_hex": "",  # TODO - Finish me
+            "second": [
+                second.mnemonic,
+                " ".join(second.registers),
+                second.parameter,
+            ],
+            "second_hex": "",  # TODO - Finish me
+        }
+
+    @property
+    def superclass_relationships(self) -> Dict[str, Set[str]]:
+        hierarchyDict = defaultdict(set)
+
+        for i in range(self.analysis.get_number_of_classes()):
+            rawClass = self.analysis.get_class_by_id(i)
+            className = rawClass.class_name.decode() + ";"
+            superclassName = (
+                rawClass.super_class.decode().replace(".", "/") + ";"
+            )
+
+            if className[0] != "L":
+                className = "L" + className
+            if superclassName[0] != "L":
+                superclassName = "L" + superclassName
+
+            hierarchyDict[className].add(superclassName)
+
+        return hierarchyDict
+
+    @property
+    def subclass_relationships(self) -> Dict[str, Set[str]]:
+        hierarchyDict = defaultdict(set)
+
+        for i in range(self.analysis.get_number_of_classes()):
+            rawClass = self.analysis.get_class_by_id(i)
+            className = rawClass.class_name.decode() + ";"
+            superclassName = (
+                rawClass.super_class.decode().replace(".", "/") + ";"
+            )
+
+            if className[0] != "L":
+                className = "L" + className
+            if superclassName[0] != "L":
+                superclassName = "L" + superclassName
+
+            hierarchyDict[superclassName].add(className)
```

Comment:
Two issues occur:
1. If dots (".") appear in the `rawClass.class_name` variable, the code will not convert them to "/".
2. If `rawClass.class_name` or `superclassName` starts with "L", the code will not add the prefix "L".

These issues will break the class name or the superclass name. For example, if `rawClass.class_name` is "L.b.clazz," the `className` will be "L.b.clazz;" instead of "LL/b/clazz;".